en/flight_modes/return.md

```diff
@@ -2,33 +2,107 @@
+
+More detailed explanations for each of the types are provided in the following sections.
+
+### Safety Point Return Type (RTL_TYPE=0) {#safety_point_return}
+
+In this return type the vehicle:
+- Ascends to a safe [return altitude](#return_altitude) (above any expected obstacles).
+- Flies to safe point (home position) via direct path
+- [Land or waits](#arrival) at descent altitude
+
+> **Note** This return type is the fallback for the other types if no mission or mission landing is defined.
+
+<span></span>
+> **Note** This return type is commonly referred to as *Return to Launch* (RTL) and *Return to Home* (RTH).
+
+
+### Mission Landing Return Type (RTL_TYPE=1) {#mission_landing_return}
+
+This return type uses a *mission landing pattern* (if defined) to set the return destination and landing behaviour.
+
+> **Note** A mission landing pattern consists of a [MAV_CMD_DO_LAND_START](https://mavlink.io/en/messages/common.html#MAV_CMD_DO_LAND_START), one or more position waypoints, and a [MAV_CMD_NAV_LAND](https://mavlink.io/en/messages/common.html#MAV_CMD_NAV_LAND).
 
-A copter/multi-rotor will first ascend to the return altitude (*by default*, [RTL_RETURN_ALT](#RTL_RETURN_ALT)) and then fly to the home position in a straight line and constant altitude (if already above the return altitude it will return at its current altitude).
+If a mission landing has been defined, the vehicle will:
+- Ascend to a safe [return altitude](#return_altitude) above any expected obstacles.
+- Fly a direct path to the start of a mission landing sequence
+- Descend/ascend to the specified waypoint altitude
+- Execute the landing sequence.
 
-When it arrives at the home/launch position it will rapidly descend to the [RTL_DESCEND_ALT](#RTL_DESCEND_ALT) altitude.
-It will then hover for the period defined in [RTL_LAND_DELAY](#RTL_LAND_DELAY) before landing.
+If no mission landing is defined PX4 will revert to a [Safety Point Return](#safety_point_return).
 
-The required return altitude can be configured using using the parameters [RTL_RETURN_ALT](#RTL_RETURN_ALT) and [RTL_CONE_ANG](#RTL_CONE_ANG) (which define a half cone centered around the home position).
+
+### Mission Path Return Type (RTL_TYPE=2) {#mission_path_return}
+
+This return type uses the mission (if defined) to provide a safe return *path*, and the mission landing pattern (if defined) to provide landing behaviour.
+If there is a mission but no mission landing pattern, the mission is flown *in reverse*.
+
+> **Note** The behaviour is fairly complex because it depends on the flight mode, and whether a mission and mission landing are defined.
+
+Mission *with* landing pattern:
+- **Mission mode:** Mission is continued in "fast-forward mode" (jumps, delay and any other non-position commands ignored, loiter and other position waypoints converted to simple waypoints) and then lands.
+- **Auto mode other than mission mode:**
```

Comment:
In non-mission auto modes you join the mission while in manual modes you just fly to the landing waypoint. FWIW that seems a bit inconsistent in that if I fly around in manual and put myself into hold mode, then my return path will be different than if I just do RTL when flying.

Reply:
@hamishwillee Yes, I agree that this can be a bit confusing. I think the main point was to make sure that if you are in manual mode but far away from the mission, which might still be on the vehicle, you don't actually try to go join the mission in order to go to the landing spot. We'd need to get fancier and actually start calculating distances to do this properly.

Reply:
Thanks for the clarification. I guess at this point just documenting what happens is a good start.